src/hotspot/share/jfr/leakprofiler/checkpoint/objectSampleWriter.cpp

```diff
@@ -625,6 +621,15 @@ ObjectSampleWriter::~ObjectSampleWriter() {
   write_array_infos(_writer);
   write_field_infos(_writer);
   write_root_descriptors(_writer);
+
+  if (field_infos != NULL) {
+    field_infos->clear();
```

Comment:
In other places, for in-place created objects, we just call the destructor manually, with a clarifying comment. I would find that a bit clearer.

Reply:
Okay. Although I don't like to call destructor directly, but there are other instances.

Reply:
> Okay. Although I don't like to call destructor directly, but there are other instances.

Yes, it looks weird, but the meaning is clearer IMHO. Thanks for taking my suggestion.